components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -309,6 +310,73 @@ void ConversationDriver::OnConversationActiveChanged(
   DVLOG(3) << "Conversation active changed: " << is_conversation_active;
   MaybeSeedOrClearSuggestions();
   MaybePopPendingRequests();
+  MaybeFetchSearchQuerySummary();
+}
+
+void ConversationDriver::ClearSearchQuerySummary() {
+  // Only clear it when the conversation only has the staged query and summary.
+  if (chat_history_.size() != 2) {
+    return;
+  }
+
+  const auto& last_turn = chat_history_.back();
+  if (last_turn->events && !last_turn->events->empty() &&
+      last_turn->events->back()->is_completion_event() &&
+      last_turn->events->back()
+          ->get_completion_event()
+          ->from_brave_search_SERP) {
+    chat_history_.clear();  // Clear the staged query and summary.
+    for (auto& obs : observers_) {
+      obs.OnHistoryUpdate();
+    }
+  }
+}
+
+void ConversationDriver::MaybeFetchSearchQuerySummary() {
+  // Clear existing search query summary if exists before fetching a new one.
+  ClearSearchQuerySummary();
+  if (!is_conversation_active_ || !HasUserOptedIn() ||
+      !IsBraveSearchSERP(GetPageURL()) || !should_send_page_contents_) {
+    // No need to fetch search query summary when any of these requirements
+    // are not met.
+    return;
+  }
+
+  FetchSearchQuerySummary(
+      base::BindOnce(&ConversationDriver::OnSearchQuerySummaryFetched,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ConversationDriver::OnSearchQuerySummaryFetched(
+    const std::optional<SearchQuerySummary>& search_query_summary) {
```

Comment:
We might have changed conversation (by navigating away) while this async operation was being performed. We should check that the `current_navigation_id_` hasn't changed (see `GetPageContent` for one of the examples where that is done).

Similarly, the user might have toggled `should_send_page_contents_`. We should check it's still `true`.

Reply:
Addressed in https://github.com/brave/brave-core/pull/25257/commits/0062fd3196588791c5ce89e58a3ec4864db4bd62.